src/AI/AIUtils.java

```diff
@@ -482,4 +506,23 @@ public int compare(Unit o1, Unit o2)
     targetLocs.remove(start); // No attacking your own position.
     return targetLocs;
   }
+
+  /**
+   * Finds allied production centers in the input set
+   */
+  public static Set<XYCoord> findAlliedIndustries(GameMap gameMap, Commander co, Iterable<XYCoord> coords, boolean ignoreMyOwn)
+  {
+    Set<XYCoord> result = new HashSet<XYCoord>();
+    for( XYCoord coord : coords )
+    {
+      Commander owner = gameMap.getLocation(coord).getOwner();
```

Comment:
I know this works because `isEnemy()` filters out null COs, but I might still prefer to have an explicit check for whether the location is actually owned.

Reply:
I feel like we'd have to change a lot of stuff if we changed how `isEnemy()` behaves.
Would a comment suffice?

Reply:
I wasn't suggesting a change to `isEnemy()` at all; rather an explicit check here instead of relying on `isEnemy()` implementation details.
It's probably not needed, really; I just didn't immediately see how "null CO" gets handled when I went through the mental checklist of edge cases.

Reply:
> I wasn't suggesting a change to `isEnemy()` at all; rather an explicit check here instead of relying on `isEnemy()` implementation details.

I'd argue that "returns true if the other CO is null" is part of the function's contract rather than an implementation detail.

> It's probably not needed, really; I just didn't immediately see how "null CO" gets handled when I went through the mental checklist of edge cases.

May as well throw in a comment, then.